src/main/controllers/AddPrivateBetaServiceController.ts

```diff
@@ -48,4 +46,22 @@ export class AddPrivateBetaServiceController extends RootController {
         });
       });
   }
+
+  private async getRolesToRegisterUser(req: AuthedRequest, allServices: Service[], serviceField: string): Promise<string[]> {
+    const selectedService = allServices.find(service => service.label === serviceField);
+    const rolesToAdd: string[] = [UserType.Citizen];
+
+    const allRoles = await  req.scope.cradle.api.getAllRoles();
+    const rolesMap = new Map(allRoles
+      .filter(role => role !== undefined)
+      .map(role => [role.id, role])
+    );
+
+    selectedService.onboardingRoles.forEach(r => {
+      if (rolesMap.has(r)) {
+        rolesToAdd.push(rolesMap.get(r).name);
+      }
+    });
```

Comment:
Should we maybe use a filter here? like we did for the roles map used in determining the assignable roles?
```suggestion
    selectedService.onboardingRoles
     .filter(r => rolesMap.has(r))
     .forEach(r => rolesToAdd.push(rolesMap.get(r).name));
```